odoo/addons/test_inherit/tests/test_uninstall.py

```diff
@@ -0,0 +1,202 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import standalone
+from odoo.tools import sql
+
+
+@standalone('inherit_uninstall')
+def test_inherited_fields_after_uninstall(env):
+    """
+    =================================
+    module: test_inherit
+
+    class TestInheritAnimal
+        _name = 'test.inherit.animal'
+
+    =================================
+    module: test_inherit_depends
+    depends: test_inherit
+
+    class TestInheritAnimal(models.Model):
+        _inherit = ['test.inherit.animal']
+
+        name = fields.Char(default='Animal')
+        height = fields.Integer()
+
+    =================================
+    module: test_inherit_inherits_depends
+    depends: test_inherit
+
+    class TestInheritBird(models.Model):
+        _name = 'test.inherit.bird'
+        _inherit = ['test.inherit.animal']
+
+        name = fields.Char(default='Bird')
+
+
+    class TestInheritOwl(models.Model):
+        _name = 'test.inherit.owl'
+        _inherit = ['test.inherit.bird]
+
+
+    class TestInheritPet(models.Model):
+        _name = 'test.inherit.pet'
+        _inherits = {'animal_id': 'test.inherit.animal'}
+
+        name = fields.Char(default='Pet')
+        animal_id = fields.Many2one('test.inherit.animal')
+
+    """
+
+    def test_fields(expected, ignore=None):
+        ignore = ignore or {}
+        for (model_name, field_name), exist in expected.items():
+            table_name = model_name.replace('.', '_')
+
+            if 'registry' not in ignore.get((model_name, field_name), ()):
+                assert (model_name in env and field_name in env[model_name]._fields) == exist
+
+            if 'column' not in ignore.get((model_name, field_name), ()):
+                assert bool(sql.column_exists(env.cr, table_name, field_name)) == exist
+
+            if 'ir.model.fields' not in ignore.get((model_name, field_name), ()):
```

Comment:
Where is this case used? I only see `registry` and `column`... maybe I'm missing something...